### PR TITLE
Jamstack chapter 2021

### PR DESCRIPTION
**Pull request by denar90**
Progress on https://github.com/HTTPArchive/almanac.httparchive.org/issues/2156

Staging url: https://20211129t003730-dot-webalmanac.uk.r.appspot.com/en/2021/jamstack
Home page featured quote: https://20211129t003730-dot-webalmanac.uk.r.appspot.com/en/2021/?feat=jamstack#featured-chapter

todo
- [x] improve texts
- [x] update some images

### CODE REVIEW

File: src/content/en/2021/jamstack.md [hunk 1]
```diff
@@ -17,4 +17,479 @@ featured_stat_3: TODO
...
+) }}
+
+The bottomline here is that Lighthouse results in general are good, but performance and PWA should be the main target for SSGs, these categories need some work to improve developer experience out of the box, hence the end result of sites performance will be improved.
+# Core Web Vitals
+
+[Core Web Vitals](https://web.dev/learn-web-vitals/) (CWV) - an initiative to provide unified guidance for quality signals that are essential to delivering a great user experience on the web. CWV itself uses 3 performance metrics:
+
+We gathered  data for SSGs and also for CDNs which deliver Jamstack sites.
+
+In overall assessment for SSGs we can understand the basic performance level of Jamstack sites. CVW assessment contains data of 75% percentile of page loads  which have a good score of CVW across all metrics.
```
Comment (Dawntraoz) [inline src/content/en/2021/jamstack.md:265]: Typo: In this case, CVW should be CWV isn't it?

**Comment by tunetheweb** [comment]
I've staged this at: https://20211129t003730-dot-webalmanac.uk.r.appspot.com/en/2021/jamstack

You can also see what the home page featured quote looks like for this chapter here: https://20211129t003730-dot-webalmanac.uk.r.appspot.com/en/2021/?feat=jamstack#featured-chapter

Updated the first comment with these links.

File: src/content/en/2021/jamstack.md [hunk 2]
```diff
@@ -1,20 +1,506 @@
...
+Worth to mention some data disclaimers:
+1. HTTPArchive data of detected SSGs is based on Wappalyzer technology. It can’t detect whether the site was built with Eleventy. Also, it can’t detect if the site was generated by Next.js [Static Rendering](https://nextjs.org/docs/basic-features/pages#static-generation-recommended) or [Server Side Rendering](https://nextjs.org/docs/basic-features/pages#server-side-rendering).
+2. In our analysis, we can’t get any info related to headless CMSs, hence we will not cover this part either.
+3. We visualise SSG data using top 5 used SSGs based on number of sites built with these SSGs.
+
+Most of the data is represented for mobile and desktop websites. There's a fully described [​​methodology](./methodology) used in HTTPArchive.
+
+## Adoption of SSGs
+
+### Ranking
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:57]: We avoid stacked headers like these. Every heading should have at least one sentence in it explaining what this section will cover.

There's a few instances of this so could you fix them all?

File: src/content/en/2021/jamstack.md [hunk 3]
```diff
@@ -1,20 +1,506 @@
...
+
+{{ figure_markup(
+  image="ssg-adoption-by-region.png",
+  caption="SSG distribution by region",
+  description="Bar chart represents SSG distribution by region. Europe has 1.1%, Americas 1.2%, Asia 1.45%, Oceania 2.19% and Africa 2% among all sites by region.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRUiAdcUCSYyS7cQcQlE9uKeMSacgRudtf4pFRkIKKCt8Nw18qOod6TaAUL7tfm8pUbCPKShEN0jJOS/pubchart?oid=805975078&format=interactive",
+  sheets_gid="1779595527",
+  sql_file="adoption_by_region.sql"
+) }}
+### Adoption by subregion
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:143]: Nit:

```suggestion

### Adoption by subregion
```

File: src/content/en/2021/jamstack.md [hunk 4]
```diff
@@ -1,20 +1,506 @@
...
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRUiAdcUCSYyS7cQcQlE9uKeMSacgRudtf4pFRkIKKCt8Nw18qOod6TaAUL7tfm8pUbCPKShEN0jJOS/pubchart?oid=180251589&format=interactive",
+  sheets_gid="1565107748",
+  sql_file="core_web_vitals_distribution.sql"
+) }}
+
+## Performance
+### Lighthouse
+
+All Lighthouse scores are median values for mobile devices. Hence, these are results from all around the world. There might be influence of using mobile data providers or devices with simpler hardware then top edge devices of 2021.
+#### Performance
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:212]: Two headings both called Performance. Avoid this.

(also stacked heading as mentioned before)

**Review by tunetheweb — CHANGES_REQUESTED** [review]
Not read it properly yet but spotted a few things for you to action.

**Comment by rviscomi** [comment]
@denar90 this is still showing as "changes requested". Is there any other feedback left to resolve or is it good for a final check before moving on to editing?

**Comment by Dawntraoz** [comment]
> @denar90 this is still showing as "changes requested". Is there any other feedback left to resolve or is it good for a final check before moving on to editing?

@rviscomi @denar90 I guess it's because 2 suggestions I made for the SEO score and Accessibility score sections. Take a look and tell me your thoughts 🤗

**Comment by denar90** [comment]
@rviscomi @Dawntraoz I've addressed all comments.

**Comment by tunetheweb** [comment]
I'm in the middle of going through this and have a few changes, if you could hold off on any more commits to this branch for now.

**Comment by tunetheweb** [comment]
Will need you to address some points in a bit @denar90 so keep a close eye for GitHub comments in next hour or so.

**Comment by denar90** [comment]
roger that :)

File: src/content/en/2021/jamstack.md [hunk 5]
```diff
@@ -1,20 +1,521 @@
...
+  caption="SSG adoption share",
+  description="Pie chart represents adoption share between SSGs. Nuxt.js covers 52.6% of Jamstack sites. Next.js is in second place with 36.8%, third one is Gatsby with 6.7%, followed by Hugo at 2.5%.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRUiAdcUCSYyS7cQcQlE9uKeMSacgRudtf4pFRkIKKCt8Nw18qOod6TaAUL7tfm8pUbCPKShEN0jJOS/pubchart?oid=2020451260&format=interactive",
+  sheets_gid="1005292860",
+  sql_file="adoption.sql"
+) }}
+
+Let's start with understanding which SSG is most popular. Nuxt.js covers 52.6% of Jamstack sites. Next.js is in second place with 36.8%, third one is Gatsby with 6.7%, followed by Hugo at 2.5%.
+
+{# TODO - add something more, rather than just repeat the description #}
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:87]: @denar90 my general feedback is a lot of the commentary on the charts are just repeating the stats. Have you anything more to say on what the results mean?

File: src/content/en/2021/jamstack.md [hunk 6]
```diff
@@ -1,20 +1,521 @@
...
+  sheets_gid="1565107748",
+  sql_file="core_web_vitals_distribution.sql"
+) }}
+
+Next.js, as most popular by data we have, mostly served by Cloudflare, Vercel, and AWS. Most of Gatsby's sites use Netlify, AWS, and Cloudflare.
+ Nuxt.js sites preferred to be served by Cloudflare, AWS, and Netlify. Hugo mostly uses Netlify, and Jekyll as no surprise mostly on GitHub.
+
+It's fair to say that in real world, Cloudflare is used not only for pure serving as CDN but as protection service, DNS balancing etc. So, keep in mind that Cloudflare results can be false positives.
+
+{# TODO Isn't this the same for Akamai? I don'y understand why that's a false positive? #}
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:208]: @denar90 ?

File: src/content/en/2021/jamstack.md [hunk 7]
```diff
@@ -1,20 +1,521 @@
...
+  caption="Median Lighthouse SEO score",
+  description="Bar chart represents median Lighthouse SEO score. All Jamstack sites provide great SEO scores. From 90 to 92.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRUiAdcUCSYyS7cQcQlE9uKeMSacgRudtf4pFRkIKKCt8Nw18qOod6TaAUL7tfm8pUbCPKShEN0jJOS/pubchart?oid=813819935&format=interactive",
+  sheets_gid="745310638",
+  sql_file="median_lighthouse_score.sql"
+) }}
+
+Simialrly, all Jamstack sites provide great SEO scores from 90 to 92.
+
+{# TODO Any thoughts on why? Is this something SSGs concentrate on more than Performance or Accessibility? #}
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:276]: @denar90

File: src/content/en/2021/jamstack.md [hunk 8]
```diff
@@ -1,20 +1,521 @@
...
+  caption="Median Lighthouse best practices score",
+  description="Bar chart represents median Lighthouse best practices score.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRUiAdcUCSYyS7cQcQlE9uKeMSacgRudtf4pFRkIKKCt8Nw18qOod6TaAUL7tfm8pUbCPKShEN0jJOS/pubchart?oid=343006289&format=interactive",
+  sheets_gid="745310638",
+  sql_file="median_lighthouse_score.sql"
+) }}
+
+All sites results mostly 87, and 93 for for some of them.
+
+{# TODO not sure this is a great stat to include since it's such and odd bucket of other audits. What is the reader supposed to take away from this? Maybe should consider dropping? #}
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:293]: Shoulld we drop the Best Practice section @denar90 ? Is it really telling us much?

File: src/content/en/2021/jamstack.md [hunk 9]
```diff
@@ -1,20 +1,521 @@
...
+  caption="Median page weight",
+  description="Bar chart represents median page weights",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRUiAdcUCSYyS7cQcQlE9uKeMSacgRudtf4pFRkIKKCt8Nw18qOod6TaAUL7tfm8pUbCPKShEN0jJOS/pubchart?oid=632409641&format=interactive",
+  sheets_gid="984097158",
+  sql_file="resource_weights.sql"
+) }}
+
+JavaScript based SSGs have almost 2x larger amount of resources then Hugo and Jekyll. Top one is ~2MB for Nuxt.js, followed by Next.js and Gatsby with almost 1.8MB and 1.7MB.
+
+{# TODO again, just reporting the facts, but any opinions as to what this means and what readers should take away from this? #}
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:419]: @denar90

File: src/content/en/2021/jamstack.md [hunk 10]
```diff
@@ -1,20 +1,521 @@
...
+
+A big surprise for us was jQuery. It wasn't a surprise that it's used for Hugo and Jekyll based sites (more than 60%), but that it's used inside React and Vue based sites wasn't expected! Next.js, Many Nuxt,js, and Gatsby sites use jQuery too.
+
+Styled-components was used for Next.js - 20% and Gatsby takes 34%% from all of third party libraries. Nuxt.js sites almost don't use it.
+
+Lodash is heavily used and was present among all SSGs up to 10% for Gatsby.
+
+Note that this data aggregates all of the pages for different sites, hence e.g. 9.5% of jQuery usage among other third parties can be used for some Gatsby sites and might not be used for other of sites at all.
+
+{# TODO I don't understand this sentence? #}
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:453]: @denar90

File: src/content/en/2021/jamstack.md [hunk 11]
```diff
@@ -1,20 +1,521 @@
...
+  caption="Median CSS weight",
+  description="Bar chart represents median CSS weights",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRUiAdcUCSYyS7cQcQlE9uKeMSacgRudtf4pFRkIKKCt8Nw18qOod6TaAUL7tfm8pUbCPKShEN0jJOS/pubchart?oid=1274478265&format=interactive",
+  sheets_gid="984097158",
+  sql_file="resource_weights.sql"
+) }}
+
+On the other hand, CSS is slightly heavier than Hugo and Jekyll.
+
+{# TODO Is this because the other sites have the CSS embeded in the JS? #}
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:469]: @denar90

File: src/content/en/2021/jamstack.md [hunk 12]
```diff
@@ -1,20 +1,521 @@
...
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRUiAdcUCSYyS7cQcQlE9uKeMSacgRudtf4pFRkIKKCt8Nw18qOod6TaAUL7tfm8pUbCPKShEN0jJOS/pubchart?oid=470820948&format=interactive",
+  sheets_gid="626634977",
+  sql_file="adoption_of_image_formats_in_ssgs.sql"
+) }}
+
+As one of the newer generation of image formats, <a hreflang="en" href="https://developers.google.com/speed/webp/">WebP</a> has 17% of usage among Jamstack sites. Compared to [last year's results](../2020/jamstack#image-formats), when WebP had only 3%, we can say it's a great improvement over one year.
+
+Still, the most used is JPEG at 29% and GIF at 27%. SVG is used on 19% of webpages.
+
+{# TODO That seems really high for GIF! Is it accurate? Any thoughts why? #}
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:503]: @denar90

File: src/content/en/2021/jamstack.md [hunk 13]
```diff
@@ -1,20 +1,521 @@
...
+
+### What the resources tells us
+
+This analysis of resource weights confirms that performance of Next.js, Nuxt.js and Gatsby are likely struggling because of huge resources. 2MB of page weight and ~700Kb of JavaScript amongst that will definitely have an impact on performance scores, especially for average mobile devices and slower networks. Heavy usage of styled-components for Next.js and Gatsby sites might be another cause (there has been research in areas of performance downgrades on this). A good signal is that image adoption of next gen image formats is growing, this should improve UX for end users in the long run.
+
+{# TODO can we link to the research? #}
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:509]: @denar90

**Review by tunetheweb — COMMENTED** [review]
@denar90 I've done an initial pass over in edit.

I've dropped the CrUX distribution charts. I understand they look pretty but, after discussion with @rviscomi , I don't think that data can be accurately relied on since we don't have a weighting in CrUX so merging across the data like that is questionable.

My general feedback on the chapter is that in a lot of places were listing the data shown in the figures, but without any of your expert opinion on what that means. Is there time to add any of that?

I also updated the ranking query to split by desktop and mobile.

I've also updated the staging link with the latest site.

**Comment by tunetheweb** [comment]
Oh and feel free to commit directly to this branch again and remove my `{# TODO ... #}` pieces when they've been addressed.

File: src/content/en/2021/jamstack.md (hunk 13 above)
Comment (denar90) [reply src/content/en/2021/jamstack.md]: added

File: src/content/en/2021/jamstack.md (hunk 12 above)
Comment (denar90) [reply src/content/en/2021/jamstack.md]: nope, last year it had lower results...

File: src/content/en/2021/jamstack.md (hunk 11 above)
Comment (denar90) [reply src/content/en/2021/jamstack.md]: added

File: src/content/en/2021/jamstack.md (hunk 10 above)
Comment (denar90) [reply src/content/en/2021/jamstack.md]: dropped

File: src/content/en/2021/jamstack.md (hunk 9 above)
Comment (denar90) [reply src/content/en/2021/jamstack.md]: updated

File: src/content/en/2021/jamstack.md (hunk 8 above)
Comment (denar90) [reply src/content/en/2021/jamstack.md]: dropped

File: src/content/en/2021/jamstack.md (hunk 7 above)
Comment (denar90) [reply src/content/en/2021/jamstack.md]: added

File: src/content/en/2021/jamstack.md (hunk 6 above)
Comment (denar90) [reply src/content/en/2021/jamstack.md]: updated

File: src/content/en/2021/jamstack.md (hunk 5 above)
Comment (denar90) [reply src/content/en/2021/jamstack.md]: updated

**Comment by denar90** [comment]
@tunetheweb updated places with new texts or dropped at all.

**Comment by tunetheweb** [comment]
> @tunetheweb updated places with new texts or dropped at all.

Thanks. Made some slight further tweaks. Will merge this when the tests complete and then can do a round of copy editing (FYI @shantsis if able to help?)

File: src/content/en/2021/jamstack.md [hunk 14]
```diff
@@ -1,20 +1,490 @@
...
+
+* Netlify
+* Vercel
+* Cloudflare
+* AWS
+* Azure
+* Akamai
+* GitHub
+
+Jamstack CDN services are not just for network delivery. They provide a lot of functionality to allow developers to easy deploy and manage Jamstack sites. For example, Netlify provide easy to use functionality to deploy sites in scope of their service so developers can just update the code and the continuous deployment process is managed for them. <a hreflang="en" href="https://bejamas.io/compare/netlify-vs-vercel/">Jamstack CDNs provide many other features<a> such as _serverless functions_, _A/B testing_ etc.
```
Comment (tunetheweb) [inline src/content/en/2021/jamstack.md:195]:
```suggestion
Jamstack CDN services are not just for network delivery. They provide a lot of functionality to allow developers to easy deploy and manage Jamstack sites. For example, Netlify provide easy to use functionality to deploy sites in scope of their service so developers can just update the code and the continuous deployment process is managed for them. <a hreflang="en" href="https://bejamas.io/compare/netlify-vs-vercel/">Jamstack CDNs provide many other features</a> such as _serverless functions_, _A/B testing_ etc.
```

**Pull request merged**